examples/01-filter/image-fft-perlin-noise.py

```diff
@@ -0,0 +1,176 @@
+that noise.
+n
+"""
+
+import numpy as np
+
+import pyvista as pv
+
+###############################################################################
+# Generate Perlin Noise
+# ~~~~~~~~~~~~~~~~~~~~~
+# Start by generating some `Perlin Noise
+# <https://en.wikipedia.org/wiki/Perlin_noise>`_ as in
+# :ref:`perlin_noise_2d_example` example.
+#
+# Note that we are generating it in a flat plane and using a frequency of 10 in
+# the x direction and 5 in the y direction. Units of the frequency is
+# ``1/pixel``.
+#
+# Also note that the dimensions of the image are a power of 2. This is because
+# the FFT is much more efficient for arrays sized as a power of 2.
+
+freq = [10, 5, 0]
+noise = pv.perlin_noise(1, freq, (0, 0, 0))
+xdim, ydim = (2**9, 2**9)
+sampled = pv.sample_function(noise, bounds=(0, 10, 0, 10, 0, 10), dim=(xdim, ydim, 1))
+
+# warp and plot the sampled noise
+warped_noise = sampled.warp_by_scalar()
+warped_noise.plot(show_scalar_bar=False, text='Perlin Noise', lighting=False)
+
+
+###############################################################################
+# Perform FFT of Perlin Noise
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Next, perform a FFT of the noise and plot the frequency content.
+# For the sake of simplicity, we will only plot the content in the first
+# quadrant.
+#
+# Note the usage of :func:`numpy.fft.fftfreq` to get the frequencies.
+
+sampled_fft = sampled.fft()
+freq = np.fft.fftfreq(sampled.dimensions[0], sampled.spacing[0])
+max_freq = freq.max()
+
+# only show the first quadrant
+subset = sampled_fft.extract_subset((0, xdim // 2, 0, ydim // 2, 0, 0))
+
+
+###############################################################################
+# Plot the Frequency Domain
+# ~~~~~~~~~~~~~~~~~~~~~~~~~
+# Now, plot the noise in the frequency domain. Note how there is more high
+# frequency content in the x direction and this matches the frequencies given
+# to :func:`pyvista.perlin_noise <pyvista.core.common_data.perlin_noise>`.
+
+# scale by log to make the plot viewable
+subset['scalars'] = np.abs(subset.active_scalars.real)
+warped_subset = subset.warp_by_scalar(factor=0.0001)
+
+pl = pv.Plotter(lighting='three lights')
+pl.add_mesh(warped_subset, cmap='blues', show_scalar_bar=False)
+pl.show_bounds(
+    axes_ranges=(0, max_freq, 0, max_freq, 0, max_freq),
+    xlabel='X Frequency',
+    ylabel='Y Frequency',
+    zlabel='Amplitude',
+    show_zlabels=False,
+    color='k',
+    font_size=26,
+)
+pl.add_text('Frequency Domain of the Perlin Noise')
+pl.show()
+
+
+###############################################################################
+# Low Pass Filter
+# ~~~~~~~~~~~~~~~
+# Let's perform a low pass filter on the frequency content and then convert it
+# back into the space (pixel) domain by immediately applying a reverse FFT.
+#
+# When converting back, keep only the real content. The imaginary content has
+# no physical meaning in the physical domain. PyVista will drop the imaginary
+# content, but will warn you of it.
```

Comment:
This is different because this is the reverse fft of the low pass and is already all real. I'm discarding the imaginary component to avoid getting warnings with our plotter (which discards complex and warns you).